.github/workflows/sync-cac-oscal.yml

```diff
@@ -0,0 +1,413 @@
+              exp: Time.now.to_i + (10 * 60), # Expires in 10 mins
+              iss: ENV["APP_ID"]
+            }
+            private_key = OpenSSL::PKey::RSA.new(ENV["PRIVATE_KEY"])
+            token = JWT.encode(payload, private_key, "RS256")
+            puts token
+          ')
+          echo "JWT_TOKEN=$JWT" >> $GITHUB_ENV
+          echo "::add-mask::$JWT" # Mask the token in logs
+        env:
+          APP_ID: ${{ secrets.APP_ID }}
+          PRIVATE_KEY: ${{ secrets.PRIVATE_KEY }}
+      # Generate the GitHub app installation access token which expires in 1 hour
+      - name: Get Installation Access Token
+        id: get-installation-token
+        run: |
+          INSTALLATION_TOKEN=$(curl -s -X POST \
+            -H "Authorization: Bearer ${{ env.JWT_TOKEN }}" \
+            -H "Accept: application/vnd.github.v3+json" \
+            https://api.github.com/app/installations/${{ secrets.INSTALLATION_ID }}/access_tokens \
+            | jq -r '.token')
+          echo "INSTALLATION_TOKEN=$INSTALLATION_TOKEN" >> $GITHUB_ENV
+          echo "::add-mask::$INSTALLATION_TOKEN" # Mask the token in logs
+      # Step 5: Detect the updates of CAC content
+      - name: Detect the changed files of the PR
+        id: changed-files
+        run: |
+          changes=("controls/" ".profile" "rule.ym" ".var")
+          response=$(gh api repos/${{ github.repository }}/pulls/${{ github.event.pull_request.number }}/files?per_page=50)
+          echo "$response" | jq -r '.[].filename' > filenames.txt
+          echo "CHANGE_FOUND=false" >> $GITHUB_ENV
+          for change in "${changes[@]}"; do
+            if grep -q "$change" filenames.txt ; then
+              echo "CHANGE_FOUND=true" >> $GITHUB_ENV
+              break
+            fi
+          done
+          cat filenames.txt
+        env:
+          GH_TOKEN: ${{ env.INSTALLATION_TOKEN }}
+      # Step 6: Setup the trestle bot environment
+      - name: Checkout OSCAL content repo
+        if: ${{ env.CHANGE_FOUND == 'true' }}
+        uses: actions/checkout@v4
+        with:
+          repository: ComplianceAsCode/oscal-content
+          path: oscal-content
+          token: ${{ env.INSTALLATION_TOKEN }}
+          fetch-depth: 0
+      - name: Checkout trestle-bot repo
+        if: ${{ env.CHANGE_FOUND == 'true' }}
+        uses: actions/checkout@v4
+        with:
+          repository: complytime/trestle-bot
+          path: trestle-bot
+      - name: Setup trestlebot
+        if: ${{ env.CHANGE_FOUND == 'true' }}
+        run: |
+          cd trestle-bot && python3 -m venv venv && source venv/bin/activate
+          python3 -m pip install --no-cache-dir "poetry==1.7.1"
+          poetry install
+      # Step 7: Get profiles, controls and level mapping
+      - name: Get profiles, controls and level mapping
+        if: ${{ env.CHANGE_FOUND == 'true' }}
+        run: |
+          cd trestle-bot && source venv/bin/activate
+          RHEL_PRODUCTS=(rhel8 rhel9 rhel10)
```

Comment:
We should also include ocp4 here and rename to "RH_PRODUCTS" in all steps using this approach.
Maybe it would be good to set the variable only once and use where necessary.